virtwho/virt/vdsm/vdsm.py

```diff
@@ -106,21 +105,22 @@ def _getLocalVdsName(self, tsPath):
 
     def _secureConnect(self):
         addr = self._getLocalVdsName(self.trust_store_path)
-
-        from M2Crypto.m2xmlrpclib import SSL_Transport
-        from M2Crypto import SSL
+        from rhsm.https import ssl
+        # It appears that SafeTransport is equivalent to M2Crypto's m2.xmlrpclib.SSL_TRANSPORT
+        # in all relevant versions
+        from six.moves.xmlrpc_client import SafeTransport
 
         KEYFILE = self.trust_store_path + '/keys/vdsmkey.pem'
         CERTFILE = self.trust_store_path + '/certs/vdsmcert.pem'
         CACERT = self.trust_store_path + '/certs/cacert.pem'
 
-        ctx = SSL.Context()
-
-        ctx.set_verify(SSL.verify_peer | SSL.verify_fail_if_no_peer_cert, 16)
-        ctx.load_verify_locations(CACERT)
-        ctx.load_cert(CERTFILE, KEYFILE)
-
-        return xmlrpclib.Server('https://%s:%s' % (addr, self.management_port), SSL_Transport(ctx))
+        ctx = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
```

Comment:
I would not recommend to use `ssl.PROTOCOL_SSLv23`. It is strongly encouraged to use: `ssl.PROTOCOL_TLSv1_2`. Is there any reason to use SSL 2/3?